windows/System/File/Platform.hsc

```diff
@@ -248,3 +260,155 @@ any_ = coerce BC.any
 
 #endif
 
+ioeSetWsPath :: IOError -> WindowsPath -> IOError
+ioeSetWsPath err =
+  ioeSetFileName err .
+  rightOrError .
+  WS.decodeWith (mkUTF16le TransliterateCodingFailure)
+
+rightOrError :: Exception e => Either e a -> a
+rightOrError (Left e)  = error (displayException e)
+rightOrError (Right a) = a
+
+-- inlined stuff from directory package
+furnishPath :: WindowsPath -> IO WindowsPath
```

Comment:
It looks like a good idea for implementing `furnishPath` would be to reuse whatever `base` does via C function here https://github.com/ghc/ghc/blob/fd40eaa17c6ce8716ec2eacc95beae194a935352/libraries/ghc-internal/src/GHC/Internal/IO/Windows/Paths.hs.

Apart from matching semantics it would mean less copying around. The only drawback is that someone will need to check with each ghc release whether the foreign function is still named the same.